File: 3566-find-the-sequence-of-strings-appeared-on-the-screen/3566-find-the-sequence-of-strings-appeared-on-the-screen.cpp
class Solution {
public:
    void solve(string ans, string target , int index , bool flag , vector<string>& answer){
        //base case
        if(ans.size()==target.size() && ans[ans.size()-1]==target[target.size()-1])return ;

        //if flag is true
        //need to add a new character 
        if(flag){
            ans.push_back('a');
            if(ans[index]==target[index]){
                answer.push_back(ans);
                solve(ans, target,index+1,flag,answer);
            }
            else{
                answer.push_back(ans);
                solve(ans, target,index,false,answer);
            }
        }
        else{
            ans[index]=ans[index]+1;
            if(ans[index]==target[index]){
                answer.push_back(ans);
                solve(ans,target,index+1,true,answer);
            }
            else{
                answer.push_back(ans);
                solve(ans,target,index,flag,answer);
            }
        }

    }
    vector<string> stringSequence(string target) {
        vector<string>answer;
        solve("",target,0,true,answer);
        return answer;
    }
};

File: 3566-find-the-sequence-of-strings-appeared-on-the-screen/3566-find-the-sequence-of-strings-appeared-on-the-screen.java
class Solution {
    List<String> answer = new ArrayList<>();
    
    public void solve(String ans, String target, int index, boolean flag) {
        // Base case
        if (ans.length() == target.length() && ans.charAt(ans.length() - 1) == target.charAt(target.length() - 1)) 
            return;
        
        // If flag is true
        if (flag) {
            ans += "a";
            if (ans.charAt(index) == target.charAt(index)) {
                answer.add(ans);
                solve(ans, target, index + 1, flag);
            } else {
                answer.add(ans);
                solve(ans, target, index, false);
            }
        } else {
            // Create new string with incremented character
            ans = ans.substring(0, index) + (char)(ans.charAt(index) + 1);
            
            if (ans.charAt(index) == target.charAt(index)) {
                answer.add(ans);
                solve(ans, target, index + 1, true);
            } else {
                answer.add(ans);
                solve(ans, target, index, flag);
            }
        }
    }
    
    public List<String> stringSequence(String target) {
        solve("", target, 0, true);
        return answer;
    }
}
